Infrastructure/Services/Repositories/GenericRepository.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Application.Interfaces.IRepositories;
+using Domain.Models;
+using Infrastructure.Data;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace Infrastructure.Services.Repositories;
+
+public class GenericRepository<T> : IGenericRepository<T> where T : class, IEntity
+{
+    protected DataContext _context;
+    private DbSet<T?> dbSet;
+    private readonly ILogger _logger;
+    
+    public GenericRepository(DataContext context, ILogger logger)
+    {
+        _context = context;
+        this.dbSet = context.Set<T>();
+        _logger = logger;
+    }
+    
+    public virtual async Task<IEnumerable<T?>> GetAll()
+    {
+        return await dbSet.ToListAsync();
+    }
+
+    public virtual async Task<T?> GetById(int id)
+    {
+        try
+        {
+            return await dbSet.FirstOrDefaultAsync(e => e.Id == id);
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Error in GetById method with id: {id}", id);
```

Comment:
Normally, an application should have centralized error handler. With it, you don't need to wrap every action in the `try...catch` block, it'll catch exceptions, write a log and return corresponding response to the client.

Reply:
i'm going to implement centralized error handler in the next iteration